.circleci/config.yml

```diff
@@ -172,6 +172,24 @@ commands:
           environment:
             FOUNDRY_PROFILE: ci
 
+  checkout-with-mise:
+    steps:
+      - checkout
+      - run:
+          name: Install mise
+          command: |
+            if command -v mise &> /dev/null; then
+              echo "mise already installed"
+            else
+              curl -sSL https://raw.githubusercontent.com/ethereum-optimism/mise/main/install.sh | bash
+            fi
+            
+            curl https://mise.run | sh
+            mise reshim
+      - run:
+          name: Install mise deps
+          command: mise install
```

Comment:
Dumb question, but could we use circle ci's caching to avoid having to actually download and install dependencies every run?

Reply:
It really depends on the amount of stuff we have to cache. CCI's network isn't great, and the cache works by tarballing everything in the target directory then shipping it to CCI's servers. The CI runner has a 10G link so it may be faster, but i want to see what the baseline is first. Main bottleneck looks to be `just` because it's a rust build - we may be able to use binaries there.